src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -1801,16 +1801,6 @@ private async Task HandleRpcAsync(JsonRpcMessage rpc)
                         catch (ObjectDisposedException)
                         {
                         }
-                        catch (Exception exception)
-                        {
-                            var e = new JsonRpcDisconnectedEventArgs(
-                                string.Format(CultureInfo.CurrentCulture, Resources.ErrorWritingJsonRpcResult, exception.GetType().Name, exception.Message),
-                                DisconnectedReason.StreamError,
-                                exception);
-
-                            // Fatal error. Raise disconnected event.
-                            this.OnJsonRpcDisconnected(e);
-                        }
```

Comment:
I think we need to bring this back. The `TransmitAsync` method only breaks the connection if the handler deemed the error fatal to the integrity of the stream. But *this* method should deem *all* errors fatal because it means the client will hang while waiting for a response.